subprojects/docs/src/docs/userguide/migrating_from_ant.adoc

```diff
@@ -0,0 +1,328 @@
+
+ * Compile classes
+ * Run tests
+ * Create JAR packages
+ * Produce Javadoc API documentation
+ * Manage dependencies
+
+There are other common tasks that depend on whether the build produces a library, an application, a webapp, and so on, for which Gradle has specific plugins.
+
+If you have such a build, we recommend that you take the approach of creating a Gradle build from scratch using the appropriate plugins — such as the Java Plugin as described in the chapter <<building_java_projects#building_java_projects,Building Java & JVM projects>> — and configuring them as necessary. You might even want to consider changing the directory structure of your project to match Gradle's conventions, e.g. `src/main/java` for Java source files, although you can usually retain the structure and inform Gradle about it via <<building_java_projects#sec:custom_java_source_set_paths,source-set configuration>>.
+
+If your build isn't a good fit for the standard plugins, then you're probably better off using `ant.importBuild()` and updating the build piecemeal to take advantage of features like incremental build. link:{javadocPath}org/gradle/api/AntBuilder.html#importBuild-java.lang.Object-org.gradle.api.Transformer-[Variant that allows you to change the name of tasks on import].
+
+Whichever approach you take, there are some common scenarios you will likely have to deal with during the migration, and we cover those in the rest of the chapter.
+
+== Managing dependencies
+
+Ant builds typically take one of two approaches to dealing with dependencies:
+
+ * Storing them with the project in a local "lib" directory
+ * Using https://ant.apache.org/ivy/[Ivy] to manage them
+
+Gradle can work with either situation with the appropriate configuration.
+
+=== Serving dependencies from a directory
+
+You can use local dependencies with Gradle in one of two ways:
+
+ * Define a <<repository_types#sec:flat_dir_resolver,flat-directory repository>> and use standard dependency declarations
+ * Attach the files directly to the appropriate dependency configurations — these are known as <<declaring_dependencies#sec:declaring_file_dependency,file dependencies>>
+
+It's easier to migrate to managed dependencies served from Maven- or Ivy-compatible repositories if you take the first approach, but doing so requires all your files to have a version in their name.
+
+To demonstrate the two techniques, consider a project that has the following library JARs in its `libs` directory:
+
+[listing]
+libs
+├── our-custom.jar
+├── log4j-1.2.8.jar
+└── commons-io-2.1.jar
+
+The custom JAR won't work with a flat-directory repository because it has no version number, but the other two JARs are fine. The following sample build script demonstrates how you can incorporate all of these libraries into a build:
+
+[source,groovy]
+----
+repositories {
```

Comment:
I think we ought to make a real `sample` out of this, with tests and Kotlin DSL sample.

Reply:
I always take this approach with first drafts so I can focus on the writing. The intention is to replace with tested samples on a subsequent pass.